app/table-streams/synonyms/index.ts

```diff
@@ -52,12 +52,13 @@ export const handler = createTriggerHandler(
             const oldestDate = min(
               synonyms.map(({ initialDate }) => initialDate)
             )!
+            const initialDate = oldestDate ? oldestDate : -1
```

Comment:
If the list is empty, then `min()` will return undefined, so you can use the nullish coalescing operator here. This will eliminate the corner case where the date is valid but 0.

```suggestion
            const initialDate = oldestDate ?? -1
```